ouroboros-consensus-cardano/test/Test/Consensus/Cardano/Golden.hs

```diff
@@ -0,0 +1,34 @@
+{-# LANGUAGE FlexibleInstances #-}
+{-# OPTIONS_GHC -Wno-orphans #-}
+module Test.Consensus.Cardano.Golden (tests) where
+
+import           System.FilePath ((</>))
+
+import           Ouroboros.Consensus.HardFork.Combinator.Serialisation
+
+import           Ouroboros.Consensus.Cardano.Block
+import           Ouroboros.Consensus.Cardano.Node
+
+import           Test.Tasty
+
+import           Test.Util.Serialisation.Golden
+
+import           Test.Consensus.Cardano.Examples
+
+tests :: TestTree
+tests = goldenTest_all codecConfig goldenDir examples
+  where
+    goldenDir = "ouroboros-consensus-cardano" </> "test" </> "golden"
+
+instance ToDirectory (HardForkNodeToNodeVersion (CardanoEras sc)) where
+  toDirectory v = case v of
+    CardanoNodeToNodeVersion1 -> "CardanoNodeToNodeVersion1"
+    CardanoNodeToNodeVersion2 -> "CardanoNodeToNodeVersion2"
```

Comment:
Check whether we don't have all combinations now

Reply:
We don't, for example `HardForkNodeToNodeDisabled (WrapNodeToNodeVersion ByronNodeToNodeVersion2)` is not a sensible combination.